app/services/metrics/review_turnaround/per_user_project.rb

```diff
@@ -3,29 +3,52 @@ module ReviewTurnaround
     class PerUserProject < BaseService
       BATCH_SIZE = 500
 
+      def initialize(interval = nil)
+        @interval = interval
+      end
+
       def call
         process
       end
 
       private
 
       def process
-        filtered_reviews_ids.each_slice(BATCH_SIZE) do |batch|
-          Events::Review.eager_load(:review_request).find(batch).lazy.each do |review|
-            entity = find_user_project(review.owner, review.pull_request.project)
-            turnaround = calculate_turnaround(review)
+        entities = []
+        ActiveRecord::Base.transaction do
+          filtered_reviews_ids.lazy.each_slice(BATCH_SIZE) do |batch|
+            Events::Review.includes(:project, :review_request, owner: :users_projects)
+                          .find(batch).each do |review|
+              entity = find_user_project(review.owner, review.project)
+              entities << entity
+              turnaround = calculate_turnaround(review)
 
-            create_metric(entity, turnaround)
+              create_or_update_metric(entity, turnaround)
+            end
           end
+          calculate_avg(entities)
         end
       end
 
       def filtered_reviews_ids
-        Events::Review.joins(:review_request, owner: :users_projects, pull_request: :project)
-                      .select('DISTINCT ON (reviews.pull_request_id) reviews.id')
-                      .where(opened_at: Time.zone.today.all_day)
+        Events::Review.where(opened_at: metric_interval)
                       .order(:pull_request_id, :opened_at)
-                      .map(&:id)
+                      .pluck(Arel.sql('DISTINCT ON (reviews.pull_request_id) reviews.id'))
+      end
+
+      def pull_requests_count_per_user_project(entities)
+        entities_count = Hash.new(0)
+        entities.map { |value| [entities_count[value] += 1, value] }
```

Comment:
I feel like the `value` variable should be `entity` in this `map`. It was hard for me to understand until I realized that :)